cmake/CheckLinuxKernel.cmake

```diff
@@ -0,0 +1,112 @@
+cmake_minimum_required(VERSION 3.10)
+
+# Retrieves the kernel version from /usr/include/linux/version.h and stores it in KERNEL_VER.
+# This should also correspond with the linux-libc version.
+# TODO: Consider other paths for linux/version.h?
+function(get_linux_libc_version_from_headers OUT_VERSION)
+    # macOS does not support these kinds of checks.
+    if (CMAKE_SYSTEM_NAME STREQUAL "Darwin")
+        return()
+    endif()
+
+    set(LINUX_VERSION_HEADER "/usr/include/linux/version.h")
+
+    if (NOT EXISTS ${LINUX_VERSION_HEADER})
+        message(ERROR "linux/version.h not found! Ensure linux-libc-dev is installed.")
+    endif()
+
+    # Read the LINUX_VERSION_CODE line
+    file(STRINGS ${LINUX_VERSION_HEADER} VERSION_LINE REGEX "#define LINUX_VERSION_CODE .*")
+
+    if (VERSION_LINE MATCHES "#define LINUX_VERSION_CODE ([0-9]+)")
+        set(LINUX_VERSION_CODE ${CMAKE_MATCH_1})
+    else()
+        message(ERROR "Failed to extract LINUX_VERSION_CODE from ${LINUX_VERSION_HEADER}")
+    endif()
+
+    # Convert LINUX_VERSION_CODE to major.minor.patch.
+    math(EXPR KERNEL_MAJOR "${LINUX_VERSION_CODE} / 65536")
+    math(EXPR KERNEL_MINOR "(${LINUX_VERSION_CODE} / 256) % 256")
+    math(EXPR KERNEL_PATCH "${LINUX_VERSION_CODE} % 256")
+
+    set(KERNEL_VERSION "${KERNEL_MAJOR}.${KERNEL_MINOR}.${KERNEL_PATCH}")
+
+    message(STATUS "Detected kernel version from headers: ${KERNEL_VERSION}")
+
+    set(${OUT_VERSION} ${KERNEL_VERSION} PARENT_SCOPE)
```

Comment:
We should just set to some false value when the header is not found or when not on Linux.